src/compile/layoutsize/parse.ts

```diff
@@ -110,20 +110,12 @@ function defaultUnitSize(model: UnitModel, sizeType: 'width' | 'height'): Layout
     const range = scaleComponent.get('range');
 
     if (hasDiscreteDomain(scaleType)) {
-      if (isVgRangeStep(range)) {
+      const size = getViewConfigDiscreteSize(config.view, sizeType);
```

Comment:
Looking at this I'm not sure why we should read size only from `getViewConfigDiscreteSize(config.view, sizeType)`.

But given we did this previously, I don't think fixing this is within the scope of this PR. 

I'm filing a separate issue (https://github.com/vega/vega-lite/issues/5226), as there might be a wrong code here.